plugins/hydra_ax_sweeper/hydra_plugins/hydra_ax_sweeper/config.py

```diff
@@ -0,0 +1,77 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+from dataclasses import dataclass, field
+from typing import Any, Dict, List, Optional
+
+from hydra.core.config_store import ConfigStore
+from hydra.types import MISSING, ObjectConf
+
+
+@dataclass
+class EarlyStopConfig:
+
+    minimize: bool = MISSING
+
+    max_epochs_without_improvement: int = 10
+
+    # An improvement larger than epsilon is considered significant
+    epsilon: float = 0.00001
+
+
+@dataclass
+class ExperimentConfig:
+
+    # Experiment name
+    name: Optional[str] = None
+
+    # Name of metric to optimize or null if you only return one metric
+    objective_name: str = "objective"
+
+    # Defaults to minimize, set to false to maximize
+    minimize: bool = True
+
+    # For the remaining paramaters, refer the Ax documentation: https://ax.dev/api/core.html#experiment
+    parameter_constraints: Optional[List[str]] = None
+    outcome_constraints: Optional[List[str]] = None
+    status_quo: Optional[Dict[str, Any]] = None
```

Comment:
are these optional or required?

Generally I see very little usage of MISSING.

Reply:
These are optional as the Ax API assigns a default value (None) to them.

Reply:
MISSING is used sparsely so that user has to override only the most important things (IMO wether to minimize or maximize). I can use it more often

Reply:
Nope, this sounds right.